### ISSUE TITLE
failed to call BPF_PROG_ATTACH (BPF_CGROUP_DEVICE, BPF_F_ALLOW_MULTI): can't attach program: invalid argument: unknown

### ISSUE THREAD

**Issue by chenhengqi** [NVIDIA/libnvidia-container#151]
I was trying to run container with cgroup v2 on kernel 5.4 (which don't have BPF_F_REPLACE), it fails with:

```shell
$ ctr run --rm --gpus 0 -t docker.io/nvidia/cuda:11.0-base cuda-11.0-base nvidia-smi
ctr: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: mount error: failed to add device rules: unable to replace original device filters program with new one: failed to call BPF_PROG_ATTACH (BPF_CGROUP_DEVICE, BPF_F_ALLOW_MULTI): can't attach program: invalid argument: unknown
```

cc @klueska

**Comment by chenhengqi** [comment]
Runc has a check against `BPF_F_REPLACE`.

https://github.com/opencontainers/runc/blob/f46b6ba2c9314cfc8caae24a32ec5fe9ef1059fe/libcontainer/cgroups/ebpf/ebpf_linux.go#L165

**Comment by klueska** [comment]
Thanks for reporting. I will make sure and address this before the next RC.

**Comment by chenhengqi** [comment]
Another problem, I didn't see any cgroup files under `/sys/fs/cgroup`.

```shell
$ ctr run --rm --gpus 0 -t docker.io/nvidia/cuda:11.0-base cuda-11.0-base bash
$ ls -alh /sys/fs/cgroup/
total 0
dr-xr-xr-x 2 root root 0 Dec 20 03:39 .
drwxr-xr-x 7 root root 0 Dec 20 03:39 ..
```

**Comment by chenhengqi** [comment]
@klueska When will the next RC be available ?

**Comment by klueska** [comment]
MR to address this issue is up:
https://gitlab.com/nvidia/container-toolkit/libnvidia-container/-/merge_requests/125

**Comment by chenhengqi** [comment]
Thanks.

Please ping me if you release a new RC, I will test it and close this issue.

**Comment by elezar** [comment]
Hi @chenhengqi. We missed updating this issue to notify you of the release of the fix which was released to stable as `v1.8.0`. We have since released `v1.8.1` of the NVIDIA Container Toolkit. Could you try updating to `v1.8.1` from our stable repos to test?

**Comment by chenhengqi** [comment]
We are now working on CentOS 8 and don't use kernel v5.4.

Will let you know if I have any trouble using the package. Thanks.
